crates/runtime/src/auth.rs

```diff
@@ -1,21 +1,55 @@
+use dirs;
+use std::fs::File;
+use std::io::Read;
+
+use serde::Deserialize;
+
 pub trait Auth {
     fn get_token(&self) -> String;
 }
 
 #[allow(clippy::module_name_repetitions)]
-pub struct AuthProviders {}
+#[derive(Deserialize)]
+pub struct AuthConfig {
+    pub auth: AuthProvider,
+}
+
+#[allow(clippy::module_name_repetitions)]
+#[derive(Deserialize)]
+pub struct AuthProvider {
+    pub provider_type: String,
+    pub key: String,
+}
 
 pub mod none;
 pub mod spiceai;
 
-impl AuthProviders {
-    #[must_use]
-    pub fn get_auth(name: &str) -> Box<dyn Auth> {
-        match name {
-            "spiceai" => {
-                todo!("SpiceAI auth not implemented yet")
-            }
-            _ => Box::new(none::NoneAuth::new()),
+#[must_use]
+pub fn get() -> Box<dyn Auth> {
```

Comment:
The reason I had the interface as `get_auth(name: &str)` was that I was expecting the runtime to load specific auth providers depending on which data sources needed it. Not a single auth object